### PR TITLE
Clean up dependencies for VirtualizedTree

### PR DESCRIPTION
**Pull request by benedikt-richter**
### Summary of changes

Clean up dependencies for VirtualizedTree.

### Context and reason for change

In order to extract a package there should be no dependencies between the VirtualizedTree and the other code.

### CODE REVIEW

File: src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx [hunk 1]
```diff
@@ -38,22 +29,23 @@ const useStyles = makeStyles({
...
-    resource: Resources | 1,
+    itemName: string,
+    item: ItemsForTree | 1,
     nodeId: string
   ) => ReactElement;
   expandedIds: Array<string>;
-  selectedResourceId: string;
-  isFileWithChildren: PathPredicate;
+  selectedItemId: string;
+  isFileWithChildren: PathPredicateForTree;
```
Comment (nicarl) [inline src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx:40]: Should be `isNodeWithChildren` ?

File: src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx [hunk 2]
```diff
@@ -38,22 +29,23 @@ const useStyles = makeStyles({
 const DEFAULT_MAX_TREE_DISPLAYED_ITEMS = 5;
 
 interface VirtualizedTreeProps {
-  resources: Resources;
+  items: ItemsForTree;
```
Comment (nicarl) [inline src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx:32]: alternative: `nodes: NodesForTree`

Comment (nicarl) [reply src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx:32]: I think we should use either `node` or `item` term consequently over all files we will extract as package. I have a preference for `node`

File: src/Frontend/Components/VirtualisedTree/List.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+function maxHeightWasGiven(
+  max: NumberOfDisplayedItemsForTree | HeightForTree
+): max is HeightForTree {
+  return Boolean((max as HeightForTree).height);
+}
+
+export function List(props: ListProps): ReactElement {
+  const classes = useStyles();
+  const cardHeight = props.cardVerticalDistance || 24;
```
Comment (nicarl) [inline src/Frontend/Components/VirtualisedTree/List.tsx:34]: 24 could be moved to a `const` in the beginning of the file

**Review by nicarl — COMMENTED** [review]
I have some general comments concerning naming. The rest looks fine

File: src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx (hunk 2 above)
Comment (benedikt-richter) [reply src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx:32]: Agree. Node sounds good to me!

Comment (benedikt-richter) [reply src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx:32]: I'll do so in follow up diff

File: src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx (hunk 1 above)
Comment (leslielazzarino) [reply src/Frontend/Components/VirtualisedTree/VirtualizedTree.tsx:40]: It is only the "resource" folders that we label as files. As now the icons are controlled from the outside we should extract this logic from the tree in a follow up pr.

File: src/Frontend/Components/VirtualisedTree/List.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    direction: 'ltr',
+  },
+});
+
+interface ListProps {
+  length: number;
+  max: NumberOfDisplayedItemsForTree | HeightForTree;
+  getListItem(index: number): ReactElement | null;
+  cardVerticalDistance?: number;
+  alwaysShowHorizontalScrollBar?: boolean;
```
Comment (leslielazzarino) [inline src/Frontend/Components/VirtualisedTree/List.tsx:24]: I guess we should either surface this property in the tree, or hard-code the option and simplify the logic.

**Review by leslielazzarino — CHANGES_REQUESTED** [review]

**Review by nicarl — APPROVED** [review]
I will check the renaming in the follow up ;-)

**Pull request merged**